test/extensions/filters/network/mysql_proxy/mysql_command_test.cc

```diff
@@ -454,13 +454,17 @@ TEST_F(MySQLCommandTest, MySQLTest20) {
   std::string command = buildAlter(TestResource::TABLE, table, "add column Id varchar (20)");
   hsql::SQLParserResult result;
   EXPECT_EQ(MYSQL_SUCCESS, encodeQuery(command, result));
-  expectStatementTypeAndTableAccessMap(result, hsql::StatementType::kStmtAlter, {});
+  expectStatementTypeAndTableAccessMap(result, hsql::StatementType::kStmtAlter,
+                                       {{table, {"alter"}}});
 }
 
 /*
  * Test query: "DROP DATABASE <DB>"
+ * Test is disabled because of a bug in SQL parsing library.
```

Comment:
wouldn't this be a regression?

Reply:
The bug was always there and this PR does not change anything. The old behaviour is preserved. The SQL parser has a bug and produces wrong output for DROP DATABASE. The old unit tests  in mysql had another bug which masked SQL parser's bug. In https://github.com/envoyproxy/envoy/blob/release/v1.14/test/extensions/filters/network/mysql_proxy/mysql_command_test.cc#L205-L209 if `expected_table_access_map` is empty, the routine does NOT check what was parser's result, which is stored in `table_access_map`. In case of DROP DATABASE the `table_access_map` is not empty, but old unit test never checked that.

I changed how parsing results are accessed and this uncovered the bug.

Reply:
then can we either delete these tests or fix them? I don't think we want permanently disabled test live in the code base.